commit.go

```diff
@@ -157,6 +173,15 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 		return "", nil, "", errors.Errorf("commit access to registry for %q is blocked by configuration", transports.ImageName(dest))
 	}
 
+	// Load the system signing policy.
+	commitPolicy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return "", nil, "", errors.Wrapf(err, "error obtaining default signature policy")
+	}
+	// Override the settings for local storage to make sure that we can always read the source "image".
+	// TODO: what to do with scopes that require signatures, which we can't read until after pushing?
+	commitPolicy.Transports[is.Transport.Name()] = storageAllowedPolicyScopes
```

Comment:
(It’s not actually documented that it is safe to modify the return value of `DefaultPolicy` — OTOH, there is no “deep copy” method, so there’s no alternative, and it is actually safe with the current implementation.

I can imagine this could break very badly if `DefaultPolicy` ever started caching the loaded policy.)

Reply:
Understood.